src/v/storage/disk_log_impl.cc

```diff
@@ -333,15 +333,39 @@ ss::future<> disk_log_impl::do_compact(compaction_config cfg) {
       "[{}] applying 'compaction' log cleanup policy with config: {}",
       config().ntp(),
       cfg);
-    // find first not compacted segment
+
+    // create a logging predicate for offsets..
+    auto offsets_compactible = [&](segment& s) {
+        if (log::has_compactible_offsets(s, cfg)) {
+            vlog(
+              gclog.debug,
```

Comment:
trace maybe, there may be hundreds of thousands of segments

Reply:
Good idea. (I think it didn't print that much spam in my 2000 partition test because it exits after it processes one segment.)

Reply:
I kept this one as `debug`, because one debug-level message each time we compact a segment is quite reasonable. The other case is trace now. .